### PR TITLE
fix jpg->png

### PR DESCRIPTION
**Pull request by Neesky**
## Motivation

When I use the htc network, a certain file name of the picture in stuffthingmaps is **xxxxjpgxxx.jpg**, and the conversion method of the original code will make it become **xxxxpngxxx.png**. I think this is extremely unreasonable, so I modified it.

## Modification

From        seg_map = img_info['filename'].replace('jpg', 'png')
To            seg_map = img_info['filename'].split(".")[0] + ".png"

### CODE REVIEW

File: mmdet/datasets/coco.py [hunk 1]
```diff
@@ -166,7 +166,7 @@ def _parse_ann_info(self, img_info, ann_info):
         else:
             gt_bboxes_ignore = np.zeros((0, 4), dtype=np.float32)
 
-        seg_map = img_info['filename'].replace('jpg', 'png')
+        seg_map = img_info['filename'].split(".")[0] + ".png"
```
Comment (hhaAndroid) [inline mmdet/datasets/coco.py:169]: Thank you for your feedback, you may also need to consider the scenario of `a.b.c.jpg`.

Comment (WhiteCipher) [reply mmdet/datasets/coco.py:169]: just use .rsplit(".", 1)[0]

Comment (Neesky) [reply mmdet/datasets/coco.py]: Thank you for your reminder and help, I have modified it.

**Review by hhaAndroid — APPROVED** [review]

**Comment by ZwwWayne** [comment]
Lint failed. Should fix it before merging.  Please click the detail here
image

**Comment by RangiLyu** [comment]
Hi, we have fixed the compatibility with MMCV in v2.18.1. Now the CI can work normally. Please pull the latest master and re-run the unit tests.
And please run `pre-commit run --all-files` to fix the lint.

**Comment by Neesky** [comment]
I had pulled the latest master, but I can't understand how to re-run the unit tests.
And I cannot run this command: `pre-commit run --all-files`. Because `bash: pre-commit: command not found `
I am new to github, sorry for so many questions .

**Comment by RangiLyu** [comment]
> I had pulled the latest master, but I can't understand how to re-run the unit tests.

Just push your commit. The GitHub action will trigger the unit tests automatically.

> And I cannot run this command: `pre-commit run --all-files`. Because `bash: pre-commit: command not found ` I am new to 
> github, sorry for so many questions .

Run `pip install pre-commit`

**Comment by Neesky** [comment]
I have pulled the latest master and re-run the unit tests.
But when I use command` pre-commit run --all-files`, A bug has occurred
```
An unexpected error has occurred: CalledProcessError: command: ('rbenv', 'init', '-')
return code: 1
expected return code: 0
stdout:
    Executable `rbenv` not found
```
Maybe it's because my current system is windows10. 
So, is the command ` pre-commit run --all-files`  necessary? What should I do to solve this problem?

**Comment by ZwwWayne** [comment]
image
Hi, the main issue is in the whitespace. You can also use https://github.com/open-mmlab/mmdetection/blob/master/.dev_scripts/linter.sh

**Comment by ZwwWayne** [comment]
Can we use img_prefix and seg_prefix to resolve this issue just like those in mmseg? see example here https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/datasets/custom.py#L82

**Comment by alfialdo** [comment]
is this issue solved?

**Comment by ZwwWayne** [comment]
Resolved in https://github.com/open-mmlab/mmdetection/pull/9078

**Comment by OpenMMLab-Assistant005** [comment]
Hi @Neesky ！We are grateful for your efforts in helping improve this open-source project during your personal time. 
Welcome to join OpenMMLab Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA
If you have a WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤

**Pull request closed without merging**